bin/setup

```diff
@@ -6,30 +6,47 @@ rescue LoadError
   fail "Please run `gem install activesupport` first"
 end
 
-mac_os = `uname`.chomp == 'Darwin'
-fail 'This setup script is only designed for Mac OS.' unless mac_os
+$os = `uname`.chomp.downcase.intern
+unless $os == :darwin || ($os == :linux && (`which apt-get` != ""))
+  fail 'This setup script is only designed for Mac OS or Debian/Ubuntu.'
+end
 
 # path to your application root.
 APP_ROOT = Pathname.new File.expand_path('../../',  __FILE__)
 
+def run_pg(cmd)
+  case $os
+  when :linux
+    cmd = cmd.gsub("\"", "\\\"") # escape quotes
+    `sudo su postgres -c "#{cmd}"`
+  when :darwn
+    `#{cmd}`
+  end
+end
+
 Dir.chdir APP_ROOT do
   # This script is a starting point to setup your application.
   # Add necessary setup steps to this file:
 
   puts '== Installing dependencies =='
 
-  if `brew ls --versions imagemagick`.present?
-    puts "imagemagick is already installed"
-  else
-    system "brew install imagemagick --with-libtiff"
-  end
-
-  %w{ postgresql solr redis go node watchman }.each do |package|
-    if `brew ls --versions #{package}`.present?
-      puts "#{package} is already installed"
+  case $os
+  when :darwin
+    if `brew ls --versions imagemagick`.present?
+      puts "imagemagick is already installed"
     else
-      system "brew install #{package}"
+      system "brew install imagemagick --with-libtiff"
+    end
+
+    %w( postgresql solr redis go node watchman qt ).each do |package|
```

Comment:
Yes, it was intentional, because I wanted to ensure that all changes I made were present on Mac & linux, and that the two paths did not drift apart. But I do not know if qt is required to build capybara-webkit on the mac. If not, qt should not be installed on the mac